src/mutation/updateItem.ts

```diff
@@ -29,20 +30,35 @@ const buildConditionExpressions = (
       const value = currentExpression.value;
 
       if (!comparator) continue;
-      const operator = keyOperatorLookup[comparator];
+      const operator = keyOperatorLookup(comparator);
       if (operator === 'BETWEEN') {
         expression += `${key} ${operator} :val${i}_1 AND :val${i}_2`;
-        attrValues[`:val${i}_1`] = { N: value[0].toString() };
-        attrValues[`:val${i}_2`] = { N: value[1].toString() };
+        attrValues[`:val${i}_1`] = marshall(value[0]);
+        attrValues[`:val${i}_2`] = marshall(value[1]);
       } else {
         expression += `${key} ${operator} :val${i}`;
-        attrValues[`:val${i}`] = { N: value.toString() };
+        attrValues[`:val${i}`] = marshall(value);
       }
     }
   }
   return { expression, attrValues };
 };
 
+const buildUpdateExpressions = (item: object) => {
+  const expressions = [];
+  const expressionValues = {};
+
+  Object.keys(item)?.forEach(key => {
+    expressions.push(`${key} = :val_${key}`);
```

Comment:
```suggestion
    expressions.push(`:key_${key} = :val_${key}`);
```

Reply:
tested out the change here and it threw a validation error cuz of reserved keywords
![image](https://user-images.githubusercontent.com/104054750/218907222-ec5e5b0c-9cf2-4260-865b-18d73319d50c.png)
Even when adding the prefix `val_`, the error is still there because the name contains the literal "store" keyword in this case.

Probably need to revise this to show something generic instead of variable names.

Reply:
Here is a version that works https://gist.github.com/shidil/cce1715ba7edcf7fe2a0274bb8aba7e1

Reply:
thanks @shidil